test/auto_parallel/pir/auto_parallel_c_embedding_pass.py

```diff
@@ -0,0 +1,207 @@
+        strategy = dist.Strategy()
+        strategy._mp_optimization.replace_with_c_embedding = use_pass
+        dist_model = dist.to_static(
+            layer, dist_loader, loss_fn, opt, strategy=strategy
+        )
+        loss_list = []
+        dist_model._engine._mode = "train"
+        dist_model.train()
+
+        if use_pass:
+            dist_program = dist_model._engine._pir_dist_main_progs["train"]
+            op_name = dist_program.global_block().ops[8].name()
+            np.testing.assert_equal(op_name, 'pd_op.c_embedding')
+
+        for epoch in range(3):
+            for batch_id, data in enumerate(dist_loader()):
+                x, label = data
+                loss = dist_model(x, label)
+                loss_list.append(loss)
+        return np.array(loss_list), dist_model
+
+    def run_dynamic(self, layer, opt, dist_loader):
+        loss_fn = nn.MSELoss()
+        loss_list = []
+        for epoch in range(3):
+            for batch_id, data in enumerate(dist_loader()):
+                x, label = data
+                out = layer(x)
+                loss = loss_fn(out, label)
+                loss_list.append(loss.numpy())
+                loss.backward()
+                opt.step()
+                opt.clear_grad()
+        return np.array(loss_list)
+
+    def test_mp_demo_net(self):
+        paddle.disable_static()
+        paddle.base.set_flags({'FLAGS_enable_pir_api': 1})
+        self.set_random_seed(self._seed)
+        data_loader = self.create_data_loader()
+        dist_dataloader = dist.shard_dataloader(
+            dataloader=data_loader,
+            meshes=[self.mesh],
+        )
+
+        self.set_random_seed(self._seed)
+        dy2static_layer_use_pass = EmbeddingNet(self.mesh)
+        dy2static_opt_use_pass = paddle.optimizer.AdamW(
+            learning_rate=0.1, parameters=dy2static_layer_use_pass.parameters()
+        )
+        dy2static_losses_use_pass, dist_model_use_pass = self.run_dy2static(
+            dy2static_layer_use_pass,
+            dy2static_opt_use_pass,
+            dist_dataloader,
+            True,
+        )
+
+        self.set_random_seed(self._seed)
+        dy2static_layer = EmbeddingNet(self.mesh)
+        dy2static_opt = paddle.optimizer.AdamW(
+            learning_rate=0.1, parameters=dy2static_layer.parameters()
+        )
+        dy2static_losses, dist_model = self.run_dy2static(
+            dy2static_layer, dy2static_opt, dist_dataloader, False
+        )
+
+        self.set_random_seed(self._seed)
+        dy_layer = CEmbeddingNet(self.mesh)
+        dy_opt = paddle.optimizer.AdamW(
+            learning_rate=0.1, parameters=dy_layer.parameters()
+        )
+        dy_losses = self.run_dynamic(dy_layer, dy_opt, data_loader)
+
+        np.testing.assert_allclose(
+            dy2static_losses_use_pass, dy2static_losses, atol=1e-7
+        )
+        np.testing.assert_allclose(
+            dy2static_losses_use_pass, dy_losses, atol=1e-7
+        )
```

Comment:
1. Require equal md5 values
2. Check the program contains `c_embedding` op when using pass